ansible/dualtor/y_cable_simulator_client.j2

```diff
@@ -19,6 +24,115 @@ helper_logger = logger.Logger(SYSLOG_IDENTIFIER)
 UPPER_TOR = "upper_tor"
 LOWER_TOR = "lower_tor"
 
+PLATFORM_JSON = 'platform.json'
+PORT_CONFIG_INI = 'port_config.ini'
+
+# A dict for mapping physical port to host interface
+g_physical_to_host_port_mapping = {}
+g_physical_to_logical_port_mapping = {}
+
+def _physical_port_to_host_port(physical_port):
+    """
+    Convert physical port to host interface index.
+    The physical port index got from spf is different with host interface index when interface is splited.
+    However, the mux bridges on vm_host are named according to host interface index (vm_topology.py).
+    So a convert is needed.
+
+    @arg physical_port: The physical port index got from sfputil
+    @returns: The host interface index (0-based)
+    """
+    global g_physical_to_host_port_mappingg
+    global g_physical_to_logical_port_mapping
+
+    if not g_physical_to_host_port_mapping or not g_physical_to_logical_port_mapping:
+        _load_port_info()
+    return g_physical_to_host_port_mapping[physical_port]
+
+def _load_port_info():
+    """
+    Parse platform.json or port_config.ini to get the mapping between physical_port and host interface index
+    and physical_port and logical port name. The mapping is saved in two global variables.
+    """
+    global g_physical_to_host_port_mapping
+    global g_physical_to_logical_port_mapping
+
+    parse_fmt_port_config_ini = False
+    parse_fmt_platform_json = False
+
+    porttabfile = device_info.get_path_to_port_config_file()
+    parse_fmt_port_config_ini = (os.path.basename(porttabfile) == PORT_CONFIG_INI)
+    parse_fmt_platform_json = (os.path.basename(porttabfile) == PLATFORM_JSON)
+
+    (platform, hwsku) = device_info.get_platform_and_hwsku()
+    if parse_fmt_platform_json:
+        ports, _, _ = get_port_config(hwsku, platform)
+        if not ports:
+            helper_logger.log_warning('Failed to get port config')
+            return
+        else:
+            logical_list = []
+            for intf in ports.keys():
+                logical_list.append(intf)
+
+            logical = natsorted(logical_list, key=lambda y: y.lower())
+            host_intf_index = 0
+            for intf_name in logical:
+                fp_port_index = -1
+                if 'index' in ports[intf_name].keys():
+                    fp_port_index = int(ports[intf_name]['index'])
+                g_physical_to_host_port_mapping[fp_port_index] = host_intf_index;
+                if g_physical_to_logical_port_mapping.get(fp_port_index) is None:
+                    g_physical_to_logical_port_mapping[fp_port_index] = [intf_name]
+                else:
+                    g_physical_to_logical_port_mapping[fp_port_index].append(intf_name)
+
+    else:
```

Comment:
I feel that the else portion could be abstracted out as a helper function. This change can come with another PR.

Reply:
Good suggestion. Updated. Thanks